collector-state/src/main/java/com/iluwatar/collectorstate/DigitMsg.java

```diff
@@ -0,0 +1,31 @@
+package com.iluwatar.collectorstate;
+
+/**
+ * class of message which contains digits.
+ */
+public class DigitMsg {
+  int pointer = 0;
+  String[] digits = new String[]{"h", "e", "l", "l", "o", "java", "design", "pattern"};
+
+  /**
+   * get digit in sequence of the message.
+
+   * @return digit
+   */
+  public String getDigit() {
+    String result = digits[pointer++];
+    return result;
+  }
+
+  /**
+   * check if the message has been read completely.
+
+   * @return true: has completed
+   */
+  public boolean isLastDigit() {
+    if (pointer == digits.length) {
+      return true;
+    }
+    return false;
+  }
+}

```

Comment:
I think it's kind of convoluted to have a single message that keeps a read pointer. We should probably move the example towards having many small stateless messages.